doobie/src/main/scala/za/co/absa/fadb/doobie/DoobieFunction.scala

```diff
@@ -90,62 +204,76 @@ private[doobie] trait DoobieFunctionWithStatus[I, R] extends DoobieFunctionBase[
  *  These classes use Doobie's `Fragment` to represent SQL queries and `DoobieEngine` to execute them.
  */
 object DoobieFunction {
+
   /**
    *  `DoobieSingleResultFunctionWithStatus` is an abstract class that extends `DBSingleResultFunctionWithStatus` with `DoobiePgEngine` as the engine type.
    *  It represents a database function that returns a single result with status.
    *
+   *  @param toFragmentsSeq a function that generates a sequence of `Fragment`s
    *  @param functionNameOverride the optional override for the function name
    *  @param schema the database schema
    *  @param dbEngine the `DoobieEngine` instance used to execute SQL queries
    *  @param readR the `Read[R]` instance used to read the query result into `R`
    *  @param readSelectWithStatus the `Read[StatusWithData[R]]` instance used to read the query result with status into `StatusWithData[R]`
    *  @tparam F the effect type, which must have an `Async` and a `Monad` instance
    */
-  abstract class DoobieSingleResultFunctionWithStatus[I, R, F[_]: Async](
+  abstract class DoobieSingleResultFunctionWithStatus[I, R, F[_]](
+    override val toFragmentsSeq: I => Seq[Fragment],
     functionNameOverride: Option[String] = None
-  )(implicit override val schema: DBSchema,
+  )(implicit
+    override val schema: DBSchema,
     val dbEngine: DoobieEngine[F],
     val readR: Read[R],
     val readSelectWithStatus: Read[StatusWithData[R]]
```

Comment:
I like the formatting of these.

Btw, some code docs have very long lines, can you improve that?

Reply:
Ok, shortened the docs.